database/gormdb_test.go

```diff
@@ -751,6 +751,54 @@ func TestGormDBUpdateCourse(t *testing.T) {
 	}
 }
 
+func TestGormDBCourseUniqueContraint(t *testing.T) {
+	// Test that a course with the same code and year cannot be created.
+	db, cleanup := qtest.TestDB(t)
+	defer cleanup()
+	wantCourse := &pb.Course{
```

Comment:
This test should probably be expanded a bit to test multiple scenarios.
- That we cannot create two courses with the same `OrganizationID` (maybe it can be done in another separate test, but it is also fine to do it as part of this test; it has to do with uniqueness.)
- That we cannot create two courses with the same `Code`-`Year` -- this was already supported.